src/xSTIR/cSTIR/cstir.cpp

```diff
@@ -1249,6 +1266,36 @@ cSTIR_computeObjectiveFunctionGradientNotDivided(void* ptr_f, void* ptr_i, int s
 	CATCH;
 }
 
+extern "C"
+void*
+cSTIR_objectiveFunctionAccumulateHessianTimesInput
+    (void* ptr_fun, void* ptr_est, void* ptr_inp, int subset, void* ptr_out)
+{
+	try {
+		ObjectiveFunction3DF& fun = objectFromHandle<ObjectiveFunction3DF>(ptr_fun);
+		STIRImageData& est = objectFromHandle<STIRImageData>(ptr_est);
+		STIRImageData& inp = objectFromHandle<STIRImageData>(ptr_inp);
+		STIRImageData& out = objectFromHandle<STIRImageData>(ptr_out);
+		Image3DF& curr_est = est.data();
+		Image3DF& input    = inp.data();
+		Image3DF& output   = out.data();
+		if (subset >= 0)
+			fun.accumulate_sub_Hessian_times_input(output, curr_est, input, subset);
+		else {
+			int nsub = fun.get_num_subsets();
+			output.fill(0.0);
+			shared_ptr<STIRImageData> sptr_sub(new STIRImageData(input));
+			Image3DF& sub = sptr_sub->data();
+			for (int s = 0; s < nsub; s++) {
+				fun.accumulate_sub_Hessian_times_input(sub, curr_est, input, subset);
+				output += sub;
+			}
```

Comment:
no need for the sum by hand. `accumulate*` automatically sums, so you can just do
```suggestion
			for (int s = 0; s < nsub; s++) {
				fun.accumulate_sub_Hessian_times_input(output, curr_est, input, subset);
```

Reply:
thanks!